examples/azure/terraform-azure-nsg-example/main.tf

```
@@ -0,0 +1,139 @@
+provider "azurerm" {
+  version = "~>2.20"
+  features {}
+}
+
+# ---------------------------------------------------------------------------------------------------------------------
+# PIN TERRAFORM VERSION TO >= 0.12
+# The examples have been upgraded to 0.12 syntax
+# ---------------------------------------------------------------------------------------------------------------------
+
+terraform {
+  required_version = ">= 0.12"
+}
+
+# ---------------------------------------------------------------------------------------------------------------------
+# DEPLOY A RESOURCE GROUP
+# See test/terraform_azure_example_test.go for how to write automated tests for this code.
+# ---------------------------------------------------------------------------------------------------------------------
+
+resource "azurerm_resource_group" "nsg_rg" {
+  name     = "${var.resource_group_name}-${var.postfix}"
+  location = var.location
+}
+
+# ---------------------------------------------------------------------------------------------------------------------
+# DEPLOY VIRTUAL NETWORK RESOURCES
+# ---------------------------------------------------------------------------------------------------------------------
+
+resource "azurerm_virtual_network" "vnet" {
+  name                = "${var.vnet_name}-${var.postfix}"
+  address_space       = ["10.0.0.0/16"]
+  location            = azurerm_resource_group.nsg_rg.location
+  resource_group_name = azurerm_resource_group.nsg_rg.name
+}
+
+resource "azurerm_subnet" "internal" {
+  name                 = "${var.subnet_name}-${var.postfix}"
+  resource_group_name  = azurerm_resource_group.nsg_rg.name
+  virtual_network_name = azurerm_virtual_network.vnet.name
+  address_prefixes     = ["10.0.17.0/24"]
+}
+
+resource "azurerm_network_interface" "main" {
+  name                = "${var.vm_nic_name}-${var.postfix}"
+  location            = azurerm_resource_group.nsg_rg.location
+  resource_group_name = azurerm_resource_group.nsg_rg.name
+
+  ip_configuration {
+    name                          = "${var.vm_nic_ip_config_name}-${var.postfix}"
+    subnet_id                     = azurerm_subnet.internal.id
+    private_ip_address_allocation = "Dynamic"
+  }
+}
+
+resource "azurerm_network_security_group" "nsg_example" {
+  name                = "${var.nsg_name}-${var.postfix}"
+  location            = azurerm_resource_group.nsg_rg.location
+  resource_group_name = azurerm_resource_group.nsg_rg.name
+}
+
+resource "azurerm_network_interface_security_group_association" "main" {
+  network_interface_id      = azurerm_network_interface.main.id
+  network_security_group_id = azurerm_network_security_group.nsg_example.id
+}
+
+resource "azurerm_network_security_rule" "allowSSH" {
```

NIT: Gruntwork style is to use underscore_case for labels. This applies to other resources that are camelCase.

```suggestion
resource "azurerm_network_security_rule" "allow_ssh" {
```

---

Updated casing to match standards.